src/perform_step/bdf_perform_step.jl

```diff
@@ -143,3 +143,73 @@ end
   end
   return
 end
+
+# QNDF1
+
+function initialize!(integrator, cache::QNDF1ConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::QNDF1ConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack uprev2,D,temp_D,D2,R,U,dtₙ₋₁,uf,κ,tol = cache
+  cnt = integrator.iter
+  if cnt == 1
+    perform_step!(integrator, cache.eulercache, repeat_step)
+    cache.uprev2 = integrator.uprev
+    cache.dtₙ₋₁ = dt
+    return
+  end
+  κ = -0.1850
+  γₖ = 1//1
+  k = 1
+  ρ = dt/dtₙ₋₁
+  D[1,1] = uprev - uprev2 # backward diff
+  temp_D .= D
+  if ρ != 1
+    U!(k,cache)
+    R!(k,ρ,cache)
+    cache.D .= D * (R * U)
```

Comment:
you don't need to do matrix multiplications. This will allocate and isn't a generally good way to write it down. Just write the update out.

Reply:
for the method of order more than one, we have to do it by mat mul?, I tried to write in way that it would be suitable for higher order method implementation.

Reply:
No it is just a small loop. Shampine wrote it as a matmul because MATLAB, but there should be a much simpler way to loop it

Reply:
Okay, I will try.

Reply:
for this method (order- one), `D`,`R` and `U` are only one element mat, so here we can use variable instead of `1*1` mat.

Reply:
Yeah that's fine